src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs

```diff
@@ -96,9 +94,24 @@ internal static ZipkinSpan ToZipkinSpan(this Activity activity, ZipkinEndpoint d
             }
 
             ZipkinEndpoint remoteEndpoint = null;
-            if ((activity.Kind == ActivityKind.Client || activity.Kind == ActivityKind.Producer) && attributeEnumerationState.RemoteEndpointServiceName != null)
+            if (activity.Kind == ActivityKind.Client || activity.Kind == ActivityKind.Producer)
             {
-                remoteEndpoint = RemoteEndpointCache.GetOrAdd(attributeEnumerationState.RemoteEndpointServiceName, ZipkinEndpoint.Create);
+                var hostNameOrIpAddress = attributeEnumerationState.HostName ?? attributeEnumerationState.IpAddress;
+
+                if ((attributeEnumerationState.RemoteEndpointServiceName == null || attributeEnumerationState.RemoteEndpointServiceNamePriority > 0)
+                    && hostNameOrIpAddress != null)
+                {
+                    var remoteEndpointStr = attributeEnumerationState.Port != default
```

Comment:
@alanwest It would be great if we didn't need this string allocation. Check out #1153. It's using ValueTuple for the dictionary key. If we made the key ValueTuple<string, int?> we might be able to avoid it?

Reply:
@CodeBlanch I'm not opposed to using a caching mechanism here to avoid string allocations, but I want to be sure to avoid any risk of unbounded memory growth of the cache. I could make a cache local to an individual export operation, but that might limit any perf gain.

It's not immediately apparent to me if the `StackExchangeRedisCallsInstrumentation` avoids unbounded memory growth. What, if anything, ever clears items from the cache?

Reply:
@alanwest The one in Redis should only be caching live traces/spans. They should remove as they complete. This cache here, will grow unbounded. It is cheating/assuming that the user will only ever connect to a reasonable set of remote services. That assumption may or may not be accurate, and we might want to add some expiration? But moving from a string key to a ValueTuple doesn't really change that 😄

Reply:
> The one in Redis should only be caching live traces/spans. They should remove as they complete.
Hmm... not seeing where this happens - I'm not very familiar with the Redis instrumentation. Though, I will parking-lot this concern since it's not related to this PR.

> It is cheating/assuming that the user will only ever connect to a reasonable set of remote services.
I think it's safer to not make this assumption. I think it is often the case that the set remote services is of a reasonable size, but I've seen numerous instances in customers applications where this set can be very large. This can be common in mutli-tenant architectures where number of hostnames can be large and hostname is used to route to a specific tenant. I think it might make sense to come back to the thought of a cache expiration in a separate issue.

> But moving from a string key to a ValueTuple doesn't really change that
Agreed, I will make this change in this PR since it seems like a solid approach.